kotlin-native/runtime/src/legacymm/cpp/Memory.cpp

```diff
@@ -252,7 +252,17 @@ class CycleDetector : private kotlin::Pinned, public KonanAllocatorAware {
 
 // TODO: can we pass this variable as an explicit argument?
 THREAD_LOCAL_VARIABLE MemoryState* memoryState = nullptr;
-THREAD_LOCAL_VARIABLE FrameOverlay* currentFrame = nullptr;
+
+#ifdef KONAN_WINDOWS
+    #include <processthreadsapi.h>
+    static DWORD currentFrameTlsIndex = TlsAlloc();
```

Comment:
What if it returns `TLS_OUT_OF_INDEXES`?

Reply:
I believe that's unlikely to happen, since this is static and it is executed once when the program starts. This would be one of the first ones to be called.
But I can wrap it into a function and make an assert or if+abort just in case

Reply:
> it is executed once when the program starts

Is this true? One can load and unload a Kotlin-produced dll dynamically.